--- NVIDIA-Linux-x86_64-340.104/kernel/nv-drm.c.orig	2017-09-15 02:16:51.000000000 +0300
+++ NVIDIA-Linux-x86_64-340.104/kernel/nv-drm.c	2017-11-13 16:02:58.307294829 +0200
@@ -14,6 +14,8 @@
 #include "os-interface.h"
 #include "nv-linux.h"
 
+#include <linux/version.h>
+
 #if defined(NV_DRM_AVAILABLE)
 
 #include <drm/drmP.h>
@@ -173,8 +175,12 @@ int __init nv_drm_init(
 {
     int ret = 0;
 #if defined(NV_DRM_AVAILABLE)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+    ret = drm_legacy_pci_init(&nv_drm_driver, pci_driver);
+#else
     ret = drm_pci_init(&nv_drm_driver, pci_driver);
 #endif
+#endif
     return ret;
 }
 
@@ -183,8 +189,12 @@ void nv_drm_exit(
 )
 {
 #if defined(NV_DRM_AVAILABLE)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+    drm_legacy_pci_exit(&nv_drm_driver, pci_driver);
+#else
     drm_pci_exit(&nv_drm_driver, pci_driver);
 #endif
+#endif
 }
 
 RM_STATUS NV_API_CALL nv_alloc_os_descriptor_handle(
